app/controllers/bookmarks_controller.rb

```diff
@@ -132,23 +132,27 @@ def status_action documents
     status = params['action']
     Array(documents.map(&:id)).each do |id|
       media_object = MediaObject.find(id)
-      if cannot? :update, media_object
-        errors += ["#{media_object.title} (#{id}) #{t('blacklight.messages.permission_denied')}."]
+      sucess = false
+      case status
```

Comment:
You could even tajke this further and put the flash logic in those if else blocks? Why do array checks later on?